platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/query/DefaultArtifactResolutionQuery.java

```diff
@@ -157,9 +146,17 @@ public ArtifactResolutionResult execute() {
             })
             .collect(Collectors.toList());
 
-        ConfigurationInternal detachedConfiguration = configurationContainer.detachedConfiguration();
-        ResolutionStrategyInternal resolutionStrategy = detachedConfiguration.getResolutionStrategy();
-        ComponentResolvers componentResolvers = ivyFactory.create(resolutionStrategy, filteredRepositories, metadataHandler.getComponentMetadataProcessorFactory(), ImmutableAttributes.EMPTY, null, attributesFactory, componentMetadataSupplierRuleExecutor);
+        ResolutionStrategyInternal resolutionStrategy = resolutionStrategyFactory.create();
+        ComponentResolvers componentResolvers = externalResolverFactory.createResolvers(
+            filteredRepositories,
+            metadataHandler.getComponentMetadataProcessorFactory(),
+            resolutionStrategy.getComponentSelection(),
```

Comment:
Add comment stating why we still use a resolution strategy here even though it is not user-configurable